biomage/experiment/download.py

```diff
@@ -0,0 +1,283 @@
+        INNER JOIN sample ON b.sample_id = sample.id ) as t"\
+    """
+
+    print(f"Querying samples for {experiment_id}...")
+
+    result_str = run_rds_command(command, SANDBOX_ID, input_env, USER, REGION, True)
+    return _process_query_output(result_str)
+
+
+def _download_samples_v2(experiment_id, input_env, output_path):
+    """
+    Download samples associated with an experiment from a given environment for v2.\n
+    """
+    bucket = f"biomage-originals-{input_env}"
+
+    samples_list = _get_samples_v2(experiment_id, input_env)
+    num_samples = len(samples_list)
+
+    print(f"{num_samples} samples found. Downloading sample files...\n")
+
+    for sample_idx, value in enumerate(samples_list.items()):
+        sample_name, samples = value
+        num_files = len(samples)
+
+        print(
+            f"Downloading files for sample {sample_name} (sample {sample_idx+1}/{num_samples})",
+        )
+
+        for file_idx, sample in enumerate(samples):
+
+            s3_path = sample["s3_path"]
+            file_name = Path(s3_path).name
+            file_path = output_path / sample_name / file_name
+
+            print(
+                f"> Downloading {sample_name}/{file_name} (file {file_idx+1}/{num_files})"
+            )
+
+            s3client = boto3.client("s3")
+            s3client.head_object(Bucket=bucket, Key=s3_path)
+
+            _download_file(bucket, s3_path, file_path)
+
+        print(f"Sample {sample_name} downloaded.\n")
+
+    _create_sample_mapping(samples_list, output_path)
+
+
+def _download_rds(experiment_id, input_env, output_path):
+    FILE_NAME = "r.rds"
+
+    bucket = f"biomage-source-{input_env}"
+    key = f"{experiment_id}/{FILE_NAME}"
+    file_path = output_path / FILE_NAME
+
+    _download_file(bucket, key, file_path)
+
+    print(f"RDS file saved to {file_path}\n")
+
+
+def _download_cellsets(experiment_id, input_env, output_path):
+    FILE_NAME = "cellsets.json"
+
+    bucket = f"cell-sets-{input_env}"
+    key = experiment_id
+    file_path = output_path / FILE_NAME
+    _download_file(bucket, key, file_path)
+    print(f"Cellsets file saved to {file_path}\n")
+
+
+@click.command()
+@click.option(
+    "-e",
+    "--experiment_id",
+    required=True,
+    help="Experiment ID to be copied.",
+)
+@click.option(
+    "-i",
+    "--input_env",
+    required=False,
+    default=PRODUCTION,
+    show_default=True,
+    help="Input environment to pull the data from.",
+)
+@click.option(
+    "-o",
+    "--output_path",
+    required=False,
+    default=None,
+    help="Name of output folder. By default this will be the experiment id.",
+)
+@click.option(
+    "-f",
+    "--files",
+    multiple=True,
+    required=False,
+    default=[SAMPLES, RDS, CELLSETS],
+    help="Files to download. By default all (samples, rds, cellsets) are downloaded.",
+)
+def download(experiment_id, input_env, output_path, files):
+    """
+    Downloads files associated with an experiment from a given environment.\n
```

Comment:
Would `downloads all sample files associated with an experiment` be more accurate?

Reply:
The function not only download sample files, but also RDS and `cellsets.json`

Reply:
yes, you are right actually